src/dodal/devices/apple2_undulator.py

```diff
@@ -312,24 +312,22 @@ class EnergyMinMax(BaseModel):
 
 
 class EnergyCoverageEntry(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True)
     Low: float
     High: float
     Poly: np.poly1d
 
-    class Config:
-        arbitrary_types_allowed = True
 
-
-class EnergyCoverage(BaseModel):
-    __root__: dict[str, EnergyCoverageEntry]
+class EnergyCoverage(RootModel):
+    root: dict[str, EnergyCoverageEntry]
 
 
 class LookupTableEntries(BaseModel):
     Energies: EnergyCoverage
     Limit: EnergyMinMax
 
 
-class Lookuptable(BaseModel):
+class Lookuptable(RootModel):
```

Comment:
blueapi uses `BaseModel`, I am not aware of the exact differences from `RootModel`, what is the reasoning here?

Reply:
I needed a custom type of 
```python 
dict[str, EnergyCoverageEntry], 
```
And RootModel just create it. 
Pre- 2.0 you can do
```python
__root__: dict[str, EnergyCoverageEntry]
```
Within Basemodel but it is now illegal.